lib/griddler/email.rb

```diff
@@ -26,6 +26,10 @@ def initialize(params)
       @raw_headers = params[:headers]
 
       @attachments = params[:attachments]
+
+      @spam_score = extract_spam_score(params[:spam_report])
```

Comment:
General feedback here: the `@spam_report` assignment could be done before this extract line. Then you could avoid shadowing the `spam_report` reader in the `extract_spam_score` method.